src/promptflow-tools/promptflow/tools/open_source_llm.py

```diff
@@ -76,6 +73,55 @@ def format_generic_response_payload(output: bytes, response_key: str) -> str:
         raise OpenSourceLLMUserError(message=message)
 
 
+def get_model_type(deployment_model: str) -> str:
+    m = re.match(r'azureml://registries/[^/]+/models/([^/]+)/versions/', deployment_model)
+    if m is None:
+        raise ValueError(f"Unexpected model format: {deployment_model}")
+    model = m[1].lower()
+    if model.startswith(ModelFamily.LLAMA.lower()):
+        return ModelFamily.LLAMA
+    elif model.startswith(ModelFamily.FALCON.lower()):
+        return ModelFamily.FALCON
+    elif model.startswith(ModelFamily.DOLLY.lower()):
+        return ModelFamily.DOLLY
+    elif model.startswith("gpt2"):
+        return ModelFamily.GPT2
+    else:
+        raise ValueError(f"Unexpected model type: {model} derived from deployed model: {deployment_model}")
+
+
+def get_deployment_from_endpoint(endpoint_name: str, deployment_name: str = None) -> Tuple[str, str, str]:
+    from promptflow._cli._utils import get_client_for_cli
```

Comment:
@chjinche is this the correct way to get the Workspace triple?

Reply:
@brynn-code @eniac871 for CLI ask.

Reply:
@gjwoods That's the function used by promptflow CLI, due to credentials and flexibilities concern, we suggest you creating MLClient directly by:
```
MLClient(
        credential=cred,
        subscription_id=subscription_id,
        resource_group_name=resource_group_name,
        workspace_name=workspace_name,
    )
```
Then you are able to control the workspace triple source, from tool input or environment variable, or anywhere else you like.
Same as the credential, you are able to control it by yourselves, we suggest using 'DefaultAzureCredential' here as it's a chained credential, it will try a series of credentials. (Note that interactive browser credential is excluded by default, if it is required to pop the auth window for user to login when no other available credentials, use it like:
```
MLClient(
        credential=DefaultAzureCredential(exclude_interactive_browser_credential=False),
        subscription_id=subscription_id,
        resource_group_name=resource_group_name,
        workspace_name=workspace_name,
    )
```

Reply:
@brynn-code, does runtime always set the ws triple in the env variables? Can we always read it from there? Is there any scenario where we would want this info to be passed down as tool input?

Reply:
for portal support, please kindly also read ws triple from env.